test/unit/repositories/test_catalogue_item.py

```diff
@@ -103,21 +109,35 @@ def test_delete(test_helpers, database_mock, catalogue_item_repository):
     test_helpers.mock_delete_one(database_mock.catalogue_items, 1)
 
     # pylint: disable=fixme
-    # TODO - (when the relevant item logic is implemented) mock it so that no children items are returned
+    # Mock `find_one` to return no child item document
+    test_helpers.mock_find_one(database_mock.items, None)
 
     catalogue_item_repository.delete(catalogue_item_id)
 
     database_mock.catalogue_items.delete_one.assert_called_once_with({"_id": CustomObjectId(catalogue_item_id)})
 
 
-def test_delete_with_children_items():
+def test_delete_with_children_items(test_helpers, database_mock, catalogue_item_repository):
     """
     Test deleting a catalogue item with children items.
 
     Verify that the `delete` method properly handles the deletion of a catalogue item with children items.
     """
-    # pylint: disable=fixme
-    # TODO - Implement this test when the relevant item logic is implemented
+    catalogue_item_id = str(ObjectId())
+
+    # Mock `find_one` to return the child item document
+    test_helpers.mock_find_one(
+        database_mock.items,
+        {
+            "_id": CustomObjectId(str(ObjectId())),
+            "catalogue_item_id": CustomObjectId(catalogue_item_id),
+            **FULL_ITEM_INFO,
+        },
+    )
+
+    with pytest.raises(ChildrenElementsExistError) as exc:
+        catalogue_item_repository.delete(catalogue_item_id)
+    assert str(exc.value) == (f"Catalogue item with ID {catalogue_item_id} has children elements and cannot be deleted")
```

Comment:
```suggestion
    assert str(exc.value) == f"Catalogue item with ID {catalogue_item_id} has children elements and cannot be deleted"
```
Looks like the formatter didn't run for some reason?